### DISCUSSION TITLE
Documentation: Icons next to components and a meaningful sort order of components

### DISCUSSION THREAD

**Discussion by iwis** [Megabit/Blazorise#2917] [Ideas]
What do you think about adding icons to components, and additionally add the option to sort components not alphabetically, but in a meaningful way as on the screenshots below. I can make the icons opensource for all component libraries, if others think that it is good idea. Some icons are missing - I haven't found a proper one, but I can draw them.
In the "Components" category there should be a divider line between "Modal" and "Alert" component.
Components in the "Components" category are in the following order:
1. Containers - starting from these concerning the whole page (layout of the whole page, menu bar, breadcrumbs, grid) and finishing on the smaller containers
2. Not-containers (alert, badge, button, figure, tooltip etc.) - sorted alphabetically

In the "Form" category, the "Validation", "Field", "Addon" components are special, thus they are at the end and have no icon to distinguish from the others.

![image](https://user-images.githubusercontent.com/8080496/133308230-7abdd10e-6750-4c7b-a342-9a7298829ec6.png)

**Comment by iwis** [comment]
I wrote about this idea also on MudBlazor GitHub: https://github.com/MudBlazor/MudBlazor/issues/1052#issuecomment-919368566

**Reply by stsrki** [reply]
Previously we had components listed in some _logical groups_. But it was confusing and a lot of people suggested to do it alphabetically. I think it is much easier now to look for a component.

Also, I don't it is a good idea to go with too many icons. They can become distracting. It's better to have icons only for the main group items. Maybe we can find better that suits the better that currently we have?

**Reply by iwis** [reply]
I wasn't sure myself if it is good idea. Maybe I will write a Tampermonkey script reordering the components if I will need it :) A Tampermonkey script should be super easy to write.
